### PR TITLE
add docs for struct scheduler

### PR DESCRIPTION
**Pull request by Datseris**

### CODE REVIEW

**Comment by Libbum** [comment]
This documentation build error is infuriating. I can't trace it.

**Comment by Libbum** [comment]
FYI: JuliaPlots/Plots.jl#2945
Coming from [this line](https://github.com/JuliaDynamics/Agents.jl/blob/39390112e79c3bf21b47e50cdac8da470c34758a/examples/diffeq.jl#L373).

**Comment by Libbum** [comment]
1.6.2 of Plots just got merged with a patch, so docs should work again now.

**Comment by Datseris** [comment]
I've re-run tests but still fail (again at the docbuiild)

**Comment by Datseris** [comment]
Nah, no, I'll try to make passing tests.

**Comment by Libbum** [comment]
It's hitting a timeout now, not the Plots error. Try a restart perhaps. Otherwise we may need to increase the `travis_wait` value.

**Comment by Datseris** [comment]
> Otherwise we may need to increase the travis_wait value

Where is this done?

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "Agents"
 uuid = "46ada45e-f475-11e8-01d0-f70cc89e6671"
 authors = ["Ali Vahdati", "George Datseris", "Tim DuBois"]
-version = "3.5"
+version = "3.5.1"
```
Comment (Libbum) [inline Project.toml:4]: I'm working on #281 at the moment. We may need to bump to 3.6 with that.

Comment (Datseris) [reply Project.toml:4]: Alright, then let's first merge #281 and then I'll rebase this one to the master

**Comment by Libbum** [comment]
Jeez. 45 minutes and the docs still fail. What's made them so heavy all of a sudden?

File: Project.toml (hunk 1 above)
Comment (Libbum) [reply Project.toml:4]: Think it'll be fine with a patch bump with this new implementation.

**Comment by Libbum** [comment]
:hankey: 

We exceed the travis limit now. We're going to have to find out whats taking so long to compile. Maybe separate it out into multiple jobs or something.

**Comment by Datseris** [comment]
Its okay, focus on #281 , I'll finish this PR once the other is done.

**Pull request merged**